### ISSUE TITLE
Audio loss of presenter after screen-share

### ISSUE THREAD

**Issue by BlurryLogic** [jitsi/jitsi-meet#10356]
### Description:

We encounter this issue in meetings with three or more participants.  After someone shares his/her screen, some other attendees lose that persons's audio. 

### Steps to reproduce:

This happens intermittently and seemingly randomly. The following steps may need to be repeated several times before seeing the issue. It may happen on the first attempt or the n-th. 

1. Create a meeting. Several participants join.
1. One user shares a video using the "Share Screen" button. 
1. The "presenting" user stops screen sharing; then the issue occurs

### Expected behavior:

* Attendees should not lose the audio of a person who was sharing her screen after the screen-sharing is stopped. 

### Actual behavior:

* Several attendees will lose the audio of the person who was screen-sharing.  The only way for an attendee to recover the audio of the former presenter is for the attendee to leave the meeting and re-join. 

### Server information:

* Jitsi Meet version: Several versions. Currently observable on meet.jit.si
* Operating System: Various/multiple

### Client information:

* Browser / app version: Various
* Operating System: Various

### Additional information:

This issue seems well described in the discussion here:  
    * https://community.jitsi.org/t/main-jitsi-meet-site-audio-lost-to-one-person-after-stopping-screen-share-started-right-before-july-10th-2021/103075 
    * The logs referred to in the above discussion are here: https://community.jitsi.org/uploads/short-url/uhTPdsiKTzGakKXAqSA42yWfqQK.txt

**Comment by saghul** [comment]
Is the user sharing their screen sharing also system audio?

**Comment by BlurryLogic** [comment]
Hi @saghul . Thanks so much for asking, and for your work in general!

I had to consult internally to get an accurate answer for you---we weren't 100% agreed how to understand what may be considered "system audio", so I'll just explain what we do and observe.   

In most cases, the testing presenter (a moderator) actually opens another tab in her browser, and loads browser content (audio + video) from a remote URL.  To present that content, she selects "Share your Screen", then selects the "Browser Tab" option, and ensures that "Share audio" is checked.  

I apologize that the above report didn't clarify that primary test scenario.  Thanks again for asking and looking into this!

**Comment by saghul** [comment]
No worries, good to know! So can you confirm that is that "share audio" is not checked things don't break?

Ping @jallamsetty1

**Comment by ashiqhassan95** [comment]
We also had the same issue. We were able to reproduce the issue and the issue occurs when screen sharing a specific tab in chrome with `Share audio` is checked. After stopping the screen share by the presenter, one of the participants cannot hear the audio of the presenter and have to rejoin the meeting.

1. Create a meeting and include 3 or more participants in the meeting.
2. Presenter starts screen share with a specific tab with `Share audio` is checked on the chrome browser.
3. After 55 minutes, the presenter stops screen sharing.
4. One of the participants in the meeting lost the audio of the presenter.

**Comment by wfleischer** [comment]
Just as reference: Seems related to what was reported a while ago on the community: [Main Jitsi meet site: Audio lost to one person after stopping screen share, started right before July 10th, 2021](https://community.jitsi.org/t/main-jitsi-meet-site-audio-lost-to-one-person-after-stopping-screen-share-started-right-before-july-10th-2021/103075)

**Comment by saghul** [comment]
Thanks for the feedback folks!

**Comment by jallamsetty1** [comment]
> We also facing the same issue. We are able to reproduce the issue and the issue occurs when screen sharing a specific tab in chrome with `Share audio` is checked. After stopping the screen share by the presenter, one of the participants cannot hear the audio of the presenter and have to rejoin the meeting.
> 
> 1. Create a meeting and include 3 or more participants in the meeting.
> 2. Presenter starts screen share with a specific tab with `Share audio` is checked on the chrome browser.
> 3. After 55 minutes, the presenter stops screen sharing.
> 4. One of the participants in the meeting lost the audio of the presenter and others can still hear the audio of the presenter.

@ashiqhassan95  Thanks for the detailed report. Will you be able to confirm if the user that has lost audio after the presenter stops sharing is a non-chrome user ?

**Comment by ashiqhassan95** [comment]
@jallamsetty1 
Sure I will notify you soon.

**Comment by BlurryLogic** [comment]
Thanks for the clarifying questions folks.  Feedback: 

RE question from @saghul :  
*  The audio loss does NOT seem to occur if we screen-share with "share audio" unchecked.  

RE question from @ashiqhassan95 : 
* The audio loss has been experienced by users on Safari and Firefox, in addition to Edge, Chrome, and jisti-meet-electron.

**Comment by ashiqhassan95** [comment]
@jallamsetty1 Issue mostly happening for the firefox users.

**Comment by jallamsetty1** [comment]
> @jallamsetty1 Issue mostly happening for the firefox users.

Thank you for confirming, this is in line with what I have been able to reproduce, will debug what is happening in this case and update here.

**Comment by Conan-Kudo** [comment]
This is happening for me as a Chrome user. @dasgeekchannel's Jitsi instance (running the latest version) has this problem.

**Comment by ashiqhassan95** [comment]
@jallamsetty1 
We were able to recreate the issue in the previous stable version (2.0.6433). Steps were following
1. User A (chrome) creates and joins the room.
2. User B (firefox) joins the room.
3. User A speaks and hears User B and vice versa.
4. User A turned on screen share using a specific tab of chrome.
5. User B cannot hear the audio of User A (presenter).

The issue was not found after upgrading our self-hosted Jitsi meet instance to the latest stable version (2.0.6689). It seems the bug has been fixed in the latest version.

Thank you.
